test/arrayops.jl

```diff
@@ -3295,3 +3295,31 @@ end
     ref = memoryref(mem, 2)
     @test parent(ref) === mem
 end
+
+@testset "Array/Memory escape analysis" begin
+    function no_allocate(T::Type{<:Union{Memory, Vector}})
+        v = T(undef, 2)
+        v[1] = 2
+        v[2] = 3
+        return v[1] + v[2]
+    end
+    function test_alloc(T; broken=false)
+        @test (@allocated no_allocate(T)) == 0 broken=broken
+    end
+    @testset "$T" for T in [Memory, Vector]
+        @testset "$ET" for ET in [Int, Union{Int, Float64}]
+            no_allocate(T{ET}) #compile
+            test_alloc(T{ET}, broken=(ET==Union{Int, Float64}) || T==Vector)
```

Comment:
Why is this broken for Vector?

Reply:
the allocation isn't getting removed for Vector with checkbounds=true (which is how the tests get run). Not really sure why, but it isn't a regression, so I think we can merge as is.